src/appimagetool_fetch_runtime.c

```diff
@@ -13,6 +15,69 @@
 bool fetch_runtime(char* arch, size_t* size, char** buffer, bool verbose) {
     // not the cleanest approach to globally init curl here, but this method shouldn't be called more than once anyway
     curl_global_init(CURL_GLOBAL_ALL);
+    CURL* handle = curl_easy_init();
+
+    if (handle == NULL) {
+        fprintf(stderr, "Failed to initialize libcurl\n");
+        curl_global_cleanup();
+        return false;
+    }
+
+    if (getenv("APPIMAGETOOL_FETCH_RUNTIME_INSECURE") != NULL) {
+        curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, 0L);
+    } else {
+        curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, 1L);
+
+        // use SSL_CERT_FILE and/or SSL_CERT_DIR if set, otherwise check for common locations
+        // https://curl.haxx.se/docs/sslcerts.html
+        // https://gitlab.com/probono/platformissues#certificates
+        // https://go.dev/src/crypto/x509/root_linux.go
+        if (getenv("SSL_CERT_FILE") != NULL) {
+            if (curl_easy_setopt(handle, CURLOPT_CAINFO, getenv("SSL_CERT_FILE")) == CURLE_OK) {
+                fprintf(stderr, "Using certificate file %s\n", getenv("SSL_CERT_FILE"));
+            }
+        } else {
+            curl_easy_setopt(handle, CURLOPT_CAINFO, NULL);
+            const char* certFiles[] = {
+            "/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
+            "/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
+            "/etc/ssl/ca-bundle.pem",                            // OpenSUSE
+            "/etc/pki/tls/cacert.pem",                           // OpenELEC
+            "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // CentOS/RHEL 7
+            "/etc/ssl/cert.pem",                                 // Alpine Linux
+            };
+            for (size_t i = 0; i < sizeof(certFiles) / sizeof(certFiles[0]); ++i) {
```

Comment:
Might be easier to just use a `NULL`-terminated list above. This works of course, but it might be more readable. You don't need any numeric index then, but can just `++` your pointer until you encounter a `NULL` value. We do this a lot in this code.